aggregators/codec.go

```diff
@@ -31,17 +36,34 @@ import (
 // ID first and then ordered by the partition ID.
 func (k *CombinedMetricsKey) MarshalBinaryToSizedBuffer(data []byte) error {
 	ivlSeconds := uint16(k.Interval.Seconds())
-	if len(data) < k.SizeBinary() {
-		return errors.New("sized buffer of insufficient length")
+	if len(data) != k.SizeBinary() {
+		return errors.New("failed to marshal due to incorrect sized buffer")
+	}
+	decodedID, err := hex.DecodeString(k.ID)
+	if err != nil {
+		return fmt.Errorf("failed to decode ID, ID must be hexadecimal string: %w", err)
+	}
```

Comment:
This doesn't feel like it belongs here. Could we make it the responsibility of the caller to decide how the ID is encoded, and just specify the size? I think we could even go as far as making the ID field of type `[16]byte`.

Reply:
Ah, good idea. I will give it a go.

Reply:
Updated.